### PR TITLE
stream-rule creation: prompt to start paused stream across all entry points

### PR DESCRIPTION
**Pull request by gally47**
## Motivation and Context

fixes https://github.com/Graylog2/graylog-plugin-enterprise/issues/13291
/nocl

## Screenshots (if appropriate):
Screenshot 2026-02-18 at 16 22 51

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have requested a documentation update.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/streamrules/StreamRulesEditor.tsx [hunk 1]
```diff
@@ -161,6 +171,13 @@ const StreamRulesEditor = ({ streamId, messageId = '', index = '' }: Props) => {
             <Button bsStyle="primary">I&apos;m done!</Button>
           </LinkContainer>
         </p>
+        <StartStreamAfterRuleCreateDialog
+          show={showStartStreamDialog}
+          streamTitle={stream.title}
```
[the remark this answers is not in the document]
Comment (gally47) [reply graylog2-web-interface/src/components/streamrules/StreamRulesEditor.tsx:175]: That is already handled in **if (!stream) {** return spinner

File: graylog2-web-interface/src/components/streamrules/CreateStreamRuleButton.tsx [hunk 2]
```diff
@@ -36,7 +34,9 @@ type Props = {
   buttonText?: string;
   className?: string;
   disabled?: boolean;
-  streamId?: string;
+  streamId: string;
```
[the remark this answers is not in the document]
Comment (gally47) [reply graylog2-web-interface/src/components/streamrules/CreateStreamRuleButton.tsx:37]: That is intentional we need **streamId** to Create Stream Rule

**Comment by laura-b-g** [comment]
# 🔍 Quick Review — PR #25052

> ⚠️ AI-generated review for additional signal. Findings are leads to verify, not approval.

## 🧭 Summary

Extracts stream-rule creation logic into a shared `useCreateStreamRule` hook and adds a `StartStreamAfterRuleCreateDialog` that prompts users to start a paused stream after creating a rule — applied consistently across all 4 stream-rule creation entry points. Also includes a minor loading-state fix in `StreamDetailsPage`.

## 📋 Issue Coverage

Linked issue: [graylog-plugin-enterprise#13291](https://github.com/Graylog2/graylog-plugin-enterprise/issues/13291)

| Requirement | Status | Notes |
|---|---|---|
| Prompt to start stream when creating rule on paused stream | ✅ Covered | `useCreateStreamRule` shows dialog when `streamIsPaused` is true |
| Entry point: More > Quick Add Rule | ✅ Covered | `StreamActions.tsx` now uses the hook |
| Entry point: More > Manage Rules | ✅ Covered | `StreamRulesEditor.tsx` now uses the hook |
| Entry point: Data Routing > Intake > Create Rule | ✅ Covered | `CreateStreamRuleButton.tsx` + `StreamRules.tsx` pass `streamIsPaused` |
| Entry point: Expanded Rules > Add Rule (streams overview) | ✅ Covered | `ExpandedRulesActions.tsx` now uses the hook |
| Dialog wording matches issue spec | ⚠️ Partial | Issue says _"Would you like to start the Stream Rules?"_ with [Start]/[Pause]; PR says _"Would you like to start stream **{title}**?"_ with [Start Stream]/[Cancel]. Intent is met but wording and button labels differ — likely intentional improvement |

## 🔧 Key Changes

- **New hook** `useCreateStreamRule.ts` — centralizes rule creation, telemetry, query invalidation, and the "start stream?" prompt logic
- **New component** `StartStreamAfterRuleCreateDialog.tsx` — thin wrapper around `ConfirmDialog` for the start-stream prompt
- **4 consumer updates** — `CreateStreamRuleButton`, `StreamRulesEditor`, `ExpandedRulesActions`, `StreamActions` all refactored to use the shared hook
- **`CreateStreamRuleButton` interface change** — `streamId` changed from optional (`string?`) to required (`string`), two new optional props added (`streamTitle`, `streamIsPaused`)
- **`StreamDetailsPage` fix** — loading check changed from `isFetching` to `!stream` (avoids re-showing spinner on background refetch)
- **New tests** — `useCreateStreamRule.test.tsx` covers rule creation + start-stream prompt flows

## ⚠️ Risk Flags

- **Breaking prop change**: `CreateStreamRuleButton.streamId` changed from optional to required. If any caller omits `streamId`, TypeScript will catch it — but worth confirming no dynamic/untyped callers exist.
- **`StreamDetailsPage` behavior change**: Switching from `isFetching` to `!stream` changes when the spinner shows. With react-query's `placeholderData`/`keepPreviousData`, this should be fine — the spinner only shows on initial load, not on background refetch. This is likely a bugfix (previously the page would flash a spinner on every refetch).

## 🚫 Blockers

- None

## 🟡 Important

- **No error notification on stream resume failure**: `onStartStream` in `useCreateStreamRule.ts` has a `try/finally` but no `catch` that shows a `UserNotification.error()`. If `StreamsStore.resume()` rejects, the dialog closes (via `finally`) but the user gets no feedback. Verify whether `StreamsStore.resume` handles its own error notifications internally — if not, a `catch` with `UserNotification.error` would improve UX.
- **No query invalidation after stream resume**: After successfully starting the stream, the stream's `disabled` status in the cache won't update until the next automatic refetch. Consider adding `queryClient.invalidateQueries({ queryKey: ['stream', streamId] })` after resume succeeds to immediately reflect the started state in the UI.

## 💡 Nice-to-have

- The `StartStreamAfterRuleCreateDialog` text says _"Stream Rules on this stream are currently paused"_ — technically **the stream** is paused (not the rules themselves). Consider: _"This stream is currently paused. New stream rules will not take effect until the stream is started."_

## ❓ Questions

- Is the `StreamDetailsPage` loading-state change (`isFetching` → `!stream`) related to this feature, or a separate bugfix that could be its own commit?
- Does `StreamsStore.resume()` already show error toasts internally, or should the hook add explicit error handling?

## ✅ Suggested Follow-ups

- **Manual verification**: Create a stream rule on a paused stream via each of the 4 entry points — confirm the dialog appears and the stream starts correctly
- **Edge case**: Create a rule on an _active_ stream — confirm no dialog appears
- **Edge case**: Open the start-stream dialog, then cancel — confirm stream remains paused
- **E2E coverage**: PR description links to [e2e-tests#1486](https://github.com/Graylog2/e2e-tests/pull/1486) — verify that PR covers the new prompt behavior

🤖 Generated by skill: graylog-pr-quick-review-summary

**Comment by laura-b-g** [comment]
# 🧪 Testing Results — PR #25052

**Verdict**: ✅ PASS · All 4 stream-rule creation entry points tested; "Start stream?" dialog works correctly on paused streams across all paths.

### 📋 Issue Coverage (Verified by Testing)

Linked issue: [graylog-plugin-enterprise#13291](https://github.com/Graylog2/graylog-plugin-enterprise/issues/13291)

| Requirement | Tested | Result |
|-------------|--------|--------|
| Prompt to start stream when creating rule on paused stream | ✅ Yes | PASS |
| Entry point: More > Quick Add Rule | ✅ Yes | PASS |
| Entry point: More > Manage Rules | ✅ Yes | PASS |
| Entry point: Data Routing > Intake > Create Rule | ✅ Yes | PASS |
| Entry point: Expanded Rules > Add Rule | ✅ Yes | PASS |
| Dialog wording and buttons | ✅ Yes | PASS |
| Start stream from dialog | ✅ Yes | PASS |
| Cancel keeps stream paused | ✅ Yes | PASS |

### ❌ Failed Assertions
- None

### 💡 Recommendations
- None

<details>
<summary>🤖 Automated (Smoke)</summary>

- ✅ PASS — 7/7 assertions passed
- Covered: Quick Add Rule on paused stream → dialog appears → Start Stream works; rule creation on running stream → no dialog

</details>

<details>
<summary>🧑‍💻 Manual Results</summary>

| Test Case | Result | Assertions | Notes |
|-----------|--------|------------|-------|
| M1: Data Routing > Intake > Create Rule (paused) | ✅ PASS | 5/5 | Dialog appeared via Stream Details entry point |
| M2: Expanded Rules > Add Rule (paused) | ✅ PASS | 5/5 | Dialog appeared via expanded rules on Streams Overview |
| M3: More > Manage Rules > Create Rule (paused) | ✅ PASS | 5/5 | Dialog appeared via StreamRulesEditor page |
| M4: Cancel on start-stream dialog | ✅ PASS | 3/3 | Stream remained paused after cancelling |
| M5: Stream Details page — no spinner flash | ✅ PASS | 2/2 | Page loads cleanly, no flash on refetch |

</details>

🤖 Generated by skill: graylog-manual-testing-assistant

File: graylog2-web-interface/src/pages/StreamDetailsPage.tsx [hunk 3]
```diff
@@ -23,9 +23,9 @@ import useStream from 'components/streams/hooks/useStream';
 
 const StreamDetailsPage = () => {
   const { streamId } = useParams<{ streamId: string }>();
-  const { data: stream, isFetching } = useStream(streamId);
+  const { data: stream } = useStream(streamId);
 
-  if (isFetching) {
+  if (!stream) {
```
Comment (laura-b-g) [inline graylog2-web-interface/src/pages/StreamDetailsPage.tsx]: This would lead to an infinite loading indicator if, for some reason, the stream was not found. Maybe return null for this case instead?

Comment (gally47) [reply graylog2-web-interface/src/pages/StreamDetailsPage.tsx]: Good points thanks, i added the fixes.

as for the error notification on failure is already handled in **StreamsStore.resume**
